sync/awsdatabricks.py

```diff
@@ -1017,71 +1047,205 @@ def _s3_contents_have_all_rollover_logs(contents: List[dict], run_end_time_secon
     )
 
 
+def _dbfs_directory_has_all_rollover_logs(contents: dict, run_end_time_millis: float):
+    files = contents["files"]
+    final_rollover_log = contents and next(
+        (
+            file
+            for file in files
+            if Path(file["path"]).stem in {"eventlog", "eventlog.gz", "eventlog.json.gz"}
+        ),
+        False,
+    )
+    return (
+        # Related to - https://docs.databricks.com/clusters/configure.html#cluster-log-delivery-1
+        # We want to make sure that the final_rollover_log has had data written to it AFTER the run has actually
+        # completed. We use this as a signal that Databricks has flushed all event log data to DBFS, and that we
+        # can proceed with the upload + prediction.
+        final_rollover_log
+        and final_rollover_log["modification_time"] >= run_end_time_millis
+    )
+
+
 def event_log_poll_duration_seconds():
     """Convenience function to aid testing"""
     return 15
 
 
-def _get_eventlog(cluster_description: dict, run_end_time_millis: int) -> Response[bytes]:
-    log_url = cluster_description.get("cluster_log_conf", {}).get("s3", {}).get("destination")
-    if log_url:
-        parsed_log_url = urlparse(log_url)
+def _get_eventlog_from_s3(
+    cluster_id: str,
+    bucket: str,
+    base_filepath: str,
+    run_end_time_millis: int,
+    poll_duration_seconds: int,
+):
```

Comment:
pulled this out of `_get_eventlog` into its own function now that we also support DBFS